### PR TITLE
Unified visualizer & override ui, enabled on all entities

### PR DESCRIPTION
**Pull request by Wumpf**
### What

Previously, we had ui for adding component overrides, under an experimental feature everywhere except the time series view + some hardcoded ui for manipulating overrides (those that were `EntityProperties` in 0.16).

This PR removes both the component override ui and the hardcoded property ui in favor of a new overview + override ui showing all currently used values by all active visualizers + ability to add/remove visualizers.

https://github.com/rerun-io/rerun/assets/1220815/425a9406-9eb2-4319-846b-fe0d0eee1a42

---

Noteworthy "sideeffects":
* `ComponentUiCallback` now uses raw arrow arrays, allowing it to show fallbacks
* kills lots'a old code :)
* `VisualizerQueryInfo::queried` keeps now archetype component ordering instead of alphabetical (not in above video yet)
* lots of inconsistency in our data flow are now visible

---

Planned direct follow-ups:
* allow various reset / override menus 
    * particularly important: remove override, add (single) override on multi-value
    * Would be great to do this with a menu button, will have to be a context menu until https://github.com/emilk/egui/issues/4607 is handled
* allow expanding to see the override/store/default/fallback stack
    * original plan was to skip on this, but the method has all the data layed out neatly now, so might as well!
* get the "Add" button integrated into the top (unless this is hard?)
*  show better docs: https://github.com/rerun-io/rerun/issues/6556
* Find a better name for the "blueprint section" below

---

Also needed but not planned short term:
* inspect multivalues
    * we could do so for store multivalues already, but everything else we can't yet select, so let's tackle this later
* ui polish

---

Part of:
* #4888

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6599?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6599?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6599)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/re_selection_panel/src/visualizer_ui.rs [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+                || !ctx.viewer_ctx.component_ui_registry.try_show_edit_ui(
+                    ctx.viewer_ctx,
+                    ui,
+                    raw_current_value,
+                    override_path,
+                    component,
+                    multiline,
+                )
+            {
+                // TODO(andreas): Unfortunately, display ui wants to do the query itself.
```
Comment (Wumpf) [inline crates/re_selection_panel/src/visualizer_ui.rs:194]: outdated comment. I fixed that by now!
(also broken comment, guess I went somewhere else halfway of writing this)

File: crates/re_selection_panel/src/visualizer_ui.rs [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+                // In fact some display UIs will struggle since they try to query additional data from the store.
+                // We pass
+                // so we have to figure out what store and path things come from.
```
Comment (Wumpf) [inline crates/re_selection_panel/src/visualizer_ui.rs:197]: broken and half-wrong comment

File: crates/re_selection_panel/src/visualizer_ui.rs [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+            .menu_button("Add", |ui| {
+                ui.style_mut().wrap_mode = Some(egui::TextWrapMode::Extend);
+                opened = true;
+
+                if visualizer_options.is_empty() {
+                    ui.close_menu();
+                }
+
+                // Present the option to add new components for each component that doesn't
+                // already have an active override.
```
Comment (jleibs) [inline crates/re_selection_panel/src/visualizer_ui.rs:314]:
```suggestion
                // Present an option to enable any visualizer that isn't already enabled.
```

**Review by jleibs — APPROVED** [review]
This is such a nice improvement!

**Pull request merged**